### PR TITLE
1538 strip credentials from agent config

### PR DESCRIPTION
**Pull request by mssalvatore**
# What does this PR do? 

Removes credentials from the config that the Agent retrieves from the Island (#1538).

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing? 
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [ ] ~Was the documentation framework updated to reflect the changes?~

## Testing Checklist

* [x] Added relevant unit tests?
* [ ] Have you successfully tested your changes locally? Elaborate:
    > Tested by running unit tests
    > Tested by using CURL to query the endpoint and inspect the output.
* [ ] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

File: monkey/infection_monkey/master/control_channel.py [hunk 1]
```diff
@@ -19,37 +19,51 @@ def __init__(self, server: str, agent_id: str):
         self._control_channel_server = server
 
     def should_agent_stop(self) -> bool:
-        if not self._control_channel_server:
-            return
-
```
Comment (VakarisZ) [inline monkey/infection_monkey/master/control_channel.py:24]: What does this have to do with credentials in config?

File: monkey/infection_monkey/master/control_channel.py [hunk 2]
```diff
@@ -19,37 +19,51 @@ def __init__(self, server: str, agent_id: str):
...
                 f"{self._control_channel_server}/api/monkey_control/{self._agent_id}",
                 verify=False,
+                proxies=ControlClient.proxies,
                 timeout=SHORT_REQUEST_TIMEOUT,
             )
 
             response = json.loads(response.content.decode())
             return response["stop_agent"]
         except Exception as e:
+            # TODO: Evaluate how this exception is handled; don't just log and ignore it.
```
Comment (VakarisZ) [inline monkey/infection_monkey/master/control_channel.py:33]: When are these todo's getting fixed? If we intend to keep them, maybe it's better to open up an issue

File: monkey/infection_monkey/master/control_channel.py (hunk 1 above)
Comment (mssalvatore) [reply monkey/infection_monkey/master/control_channel.py:24]: Nothing, but it's a minor fix that I don't think distracts from the purpose of this PR.

File: monkey/monkey_island/cc/services/config.py [hunk 3]
```diff
@@ -425,3 +426,20 @@ def get_config_propagation_credentials():
...
+
+    @staticmethod
+    def _remove_credentials_from_flat_config(config: Dict):
+        fields_to_remove = {
+            "exploit_lm_hash_list",
+            "exploit_ntlm_hash_list",
+            "exploit_password_list",
+            "exploit_ssh_keys",
+            "exploit_user_list",
+        }
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/services/config.py:442]: Probably would be better to not pull them from the database: https://docs.mongodb.com/manual/tutorial/project-fields-from-query-results/#return-all-but-the-excluded-fields

Not 100% sure we can do it with current pymongo

File: monkey/infection_monkey/master/control_channel.py (hunk 2 above)
Comment (mssalvatore) [reply monkey/infection_monkey/master/control_channel.py:33]: I'm planning to address these during the implementation of the AutomatedMaster.

File: monkey/monkey_island/cc/services/config.py (hunk 3 above)
Comment (mssalvatore) [reply monkey/monkey_island/cc/services/config.py:442]: Agreed, but some other reformatting needs to be done to support the agent work anyway. And whatever we do here will probably have to be redone when we refactor and add a DAL. Additionally, we need to support the old and new configuration formats for a while until we refactor our exploiters, so we'd need two different mongo queries to support both formats.

**Pull request merged**